src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -464,10 +465,7 @@ export async function sendTelemetryForPythonKernelExecutable(
         }
         const sysExecutable = concatMultilineString(output.text).trim().toLowerCase();
         const match = kernelConnection.interpreter.path.toLowerCase() === sysExecutable;
-        sendTelemetryEvent(Telemetry.PythonKerneExecutableMatches, undefined, {
-            match: match ? 'true' : 'false',
-            kernelConnectionType: kernelConnection.kind
-        });
+        trackKernelResourceInformation(Uri.file(file), { interpreterMatchesKernel: match });
```

Comment:
With the existing infrastructure we can tell:
* If its a notebook vs interactive window
* If its custom vs native notebook
* If user ran the cell vs auto start
All three are crucial and this wasn't easy to get with the previous event.

Removing the old item, once we add this, we can deprecate the old event.